app/webpacker/components/RegistrationsV2/Register/StripeWrapper.jsx

```diff
@@ -1,80 +1,78 @@
-    isLoading: isPaymentIdLoading,
-    isError,
-  } = useQuery({
-    queryKey: ['payment-secret', competitionInfo.id],
-    queryFn: () => getPaymentId(competitionInfo.id),
-    refetchOnWindowFocus: false,
-    refetchOnReconnect: false,
-    staleTime: Infinity,
-    refetchOnMount: 'always',
-    onError: (data) => {
-      const { error } = data.json;
-      dispatch(setMessage(
-        error
-          ? I18n.t(`competitions.registration_v2.errors.${error}`)
-          : I18n.t('registrations.flash.failed') + data.message,
-        'negative',
-      ));
-    },
-  });
+const convertISOAmount = async (amount, currency) => {
+  const { data } = await fetchJsonOrError(
+    paymentDenominationUrl(amount, currency),
+  );
+  return data;
+};
 
-  const { data: config, isLoading: isConfigLoading } = useQuery({
-    queryKey: ['payment-config', competitionInfo.id, paymentInfo?.id],
-    queryFn: () => getStripeConfig(competitionInfo.id, paymentInfo?.id),
-    onError: (err) => setMessage(err.error, 'error'),
-    enabled:
-      !isPaymentIdLoading && !isError && paymentInfo?.status !== 'succeeded',
-    refetchOnWindowFocus: false,
-    refetchOnReconnect: false,
-    staleTime: Infinity,
-    refetchOnMount: 'always',
-  });
+export default function StripeWrapper({
+  competitionInfo,
+  stripePublishableKey,
+  connectedAccountId,
+  user,
+  registration,
+  nextStep,
+}) {
+  const [stripePromise, setStripePromise] = useState(null);
+  const initialAmount = competitionInfo.base_entry_fee_lowest_denomination;
+  const [amount, setAmount] = useState(initialAmount);
+  const [donationAmount, setDonationAmount] = useState(0);
+  const [displayAmount, setDisplayAmount] = useState();
 
   useEffect(() => {
-    if (!isConfigLoading) {
-      setStripePromise(
-        loadStripe(config.stripe_publishable_key, {
-          stripeAccount: config.connected_account_id,
-        }),
+    async function fetchInitialAmount() {
+      const { human_amount: humanAmount } = await convertISOAmount(
+        initialAmount,
+        competitionInfo.currency_code,
       );
+      setDisplayAmount(humanAmount);
     }
-  }, [
-    config?.connected_account_id,
-    config?.stripe_publishable_key,
-    isConfigLoading,
-  ]);
+    fetchInitialAmount();
+  }, [competitionInfo.currency_code, initialAmount]);
+
+  useEffect(() => {
+    setStripePromise(
+      loadStripe(stripePublishableKey, {
+        stripeAccount: connectedAccountId,
+      }),
+    );
+  }, [connectedAccountId, stripePublishableKey]);
+  const handleDonation = useCallback(async (_, { value: newDonationAmount }) => {
+    const {
+      api_amounts: { stripe: stripeAmount },
+      human_amount: humanAmount,
+    } = await convertISOAmount(initialAmount + newDonationAmount, competitionInfo.currency_code);
+    setAmount(stripeAmount);
+    setDonationAmount(newDonationAmount);
+    setDisplayAmount(humanAmount);
+  }, [competitionInfo.currency_code, initialAmount]);
```

Comment:
I'm a bit unhappy about this `async`-heavy handling of the API call. The biggest problem is that the UI stays interactive while the new amount is loading.

If there is a popular competition, it may be conceivable that psych sheets or whatever else is clogging our infrastructure, and the return value from the conversion API might arrive late. The user types in their credit card details first, then at the very last moment while hovering over the "Pay Now" button sees the donation field, and spontaneously decides to give five dollars.
They then proceed to click the button right away (because they had already entered their credit card details before) but are never actually donating anything because our API didn't return the converted currency amount in time.